### PR TITLE
Added Notifications

### PR DESCRIPTION
**Pull request by zhenlineo**

### CODE REVIEW

**Comment by zhenlineo** [comment]
Pending: https://github.com/neo4j/neo4j/pull/5752

File: driver/src/main/java/org/neo4j/driver/InputPosition.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.neo4j.driver;
+
+/**
+ * An input position refers to a specific point in a query string.
```
Comment (jakewins) [inline driver/src/main/java/org/neo4j/driver/InputPosition.java]: `s/point in a query string/character in a statement/g`

File: driver/src/main/java/org/neo4j/driver/Notification.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.neo4j.driver;
+
+/**
+ * Representation for notifications found when executing a query.
```
Comment (jakewins) [inline driver/src/main/java/org/neo4j/driver/Notification.java]: `s/query/statement/g`

File: driver/src/main/java/org/neo4j/driver/ResultSummary.java [hunk 3]
```diff
@@ -73,4 +75,14 @@
      * @return profiled statement plan for the executed statement if available
      */
     ProfiledPlan profile();
+
+    /**
+     * A list of notifications that might arise when executing the statement.
+     * Notifications can be warnings about problematic queries or other valuable information that can be presented
```
Comment (jakewins) [inline driver/src/main/java/org/neo4j/driver/ResultSummary.java]: query -> statement :)

**Comment by jakewins** [comment]
Looks awesome!

Somewhat unrelated: I'm wondering if we (as part of another PR) should move the result-summary related things into their own package (like `summary` or something), the main API package is getting a bit crowded. I guess it's not a huge issue as-is..

**Comment by boggle** [comment]
@jakewins: +1 on that

**Comment by zhenlineo** [comment]
I am not sure how do you suggest to update the main API package. I pulled some inner classes into a inner.summary package. Is that the refactoring you want? @jakewins @boggle

**Pull request merged**